packages/context/test/acceptance/extension-point.ts

```diff
@@ -0,0 +1,215 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {Context, Binding, BindingScope, inject} from '../..';
+
+// tslint:disable:no-any
+/**
+ * Interface for the extension point configuration
+ */
+export interface ExtensionPointConfig {
+  extensions: {
+    [extensionName: string]: any;
+  };
+  extensionPoint: {
+    [property: string]: any;
+  };
+}
+
+/**
+ * Base class for extension points
+ */
+export abstract class ExtensionPoint<EXT extends object> {
+  /**
+   * Configuration (typically to be injected)
+   */
+  protected configuration: ExtensionPointConfig;
+
+  constructor(
+    /**
+     * The unique name of this extension point. It also serves as the binding
+     * key prefix for bound extensions
+     */
+    protected name: string,
+    /**
+     * The Context (typically to be injected)
+     */
+    protected context: Context,
+    /**
+     * Configuration (typically to be injected)
+     */
+    configuration?: ExtensionPointConfig,
+  ) {
+    this.configuration = configuration || {extensionPoint: {}, extensions: {}};
+  }
+
+  /**
+   * Find an array of bindings for extensions
+   */
+  getExtensionBindings(): Binding[] {
```

Comment:
The difference between names `getExtensionBindings` and `getExtensionBinding` (below) is too small and easy to overlook. I am proposing to rename this one to `getAllExtensionBindings`.

Reply:
+1.